### PR TITLE
mpir: MPI_Comm_dup_with_info: added support of user-provided hints 

### PR DESCRIPTION
**Pull request by alexander-sannikov**

### CODE REVIEW

File: src/mpi/comm/commutil.c [hunk 1]
```diff
@@ -546,6 +550,17 @@ int MPII_Comm_is_node_consecutive(MPIR_Comm * comm)
...
+{
+    return MPII_Comm_copy_with_info(comm_ptr, size, outcomm_ptr, NULL);
+}
+
+
+#undef FUNCNAME
+#define FUNCNAME MPII_Comm_copy_with_info
+#undef FCNAME
+#define FCNAME "MPII_Comm_copy_with_info"
+int MPII_Comm_copy_with_info(MPIR_Comm * comm_ptr, int size, MPIR_Comm ** outcomm_ptr,
```
Comment (raffenet) [inline src/mpi/comm/commutil.c]: TBH, I'd rather just add an info parameter to `MPII_Comm_copy` than add this extra layer. What do you think @alexander-sannikov?

Comment (alexander-sannikov) [reply src/mpi/comm/commutil.c]: @raffenet totally agree.

**Comment by alexander-sannikov** [comment]
@raffenet may we merge this?

**Comment by sagarth** [comment]
We had offline discussions with @alexander-sannikov and @raffenet . We decided to hold this PR for now. Soon we will submit another PR to migrate comm hints from MPIR-based to MPID- based infrastructure. At that point, we will discuss about the two PRs together.

**Comment by wesbland** [comment]
I believe the new PR from @sagarth will replace this one. Closing.

**Pull request closed without merging**